pkg/config/config_template.yaml

```diff
@@ -324,6 +324,21 @@ api_key:
 # Duration of the restart interval in seconds
 # jmx_restart_interval: 5
 #
+# JMXFetch now collects multiples instances concurrently. The following metrics may
+# help fine-tune the level of concurrency and timeouts that come into play during the
+# collection of metrics from configured instances:
+#
+# thread_pool_size: determines the maximum level of concurrency.
```

Comment:
nit: IMO it's a bit weird to write `thread_pool_size` and prefix it with `jmx_` below. Just a description like `# Maximum level of concurrency` would be better I think (ideally with a short explanation of what the user-facing tradeoff would be when it's increased/lowered)

Reply:
(applies to the other options as well)